airflow/gcp/hooks/gsheets.py

```diff
@@ -0,0 +1,367 @@
+# -*- coding: utf-8 -*-
+#
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+#
+"""
+This module contains a Google Sheets API hook
+"""
+
+from googleapiclient.discovery import build
+from airflow.contrib.hooks.gcp_api_base_hook import GoogleCloudBaseHook
+from airflow.exceptions import AirflowException
+
+
+class GSheetsHook(GoogleCloudBaseHook):
+    """
+    Interact with Google Sheets via GCP connection
+    Reading and writing cells in Google Sheet:
+    https://developers.google.com/sheets/api/guides/values
+
+    :param gcp_conn_id: The connection ID to use when fetching connection info.
+    :type gcp_conn_id: str
+    :param spreadsheet_id: The Google Sheet ID to interact with
+    :type spreadsheet_id: str
+    :param api_version: API Version
+    :type api_version: str
+    :param delegate_to: The account to impersonate, if any.
+        For this to work, the service account making the request must have
+        domain-wide delegation enabled.
+    :type delegate_to: str
+    """
+
+    def __init__(
+        self,
+        spreadsheet_id: str,
+        gcp_conn_id: str = 'google_cloud_default',
+        api_version: str = 'v4',
+        delegate_to: str = None
+    ) -> None:
+        super().__init__(gcp_conn_id, delegate_to)
+        self.spreadsheet_id = spreadsheet_id
+        self.gcp_conn_id = gcp_conn_id
+        self.api_version = api_version
+        self.delegate_to = delegate_to
+        self.num_retries = self._get_field('num_retries', 5)
+        self._conn = None
+
+    def get_conn(self):
```

Comment:
```suggestion
    def get_conn(self) -> Any:
```
This is not the best type, but we do not have a better one. 
Reference: